src/exchange_adapters/kraken.ts

```diff
@@ -0,0 +1,84 @@
+import { Exchange } from '../utils'
+import { BaseExchangeAdapter, ExchangeAdapter, ExchangeDataType, Ticker, Trade } from './base'
+
+export class KrakenAdapter extends BaseExchangeAdapter implements ExchangeAdapter {
+         baseApiUrl = 'https://api.kraken.com'
+         readonly _exchangeName = Exchange.KRAKEN
+
+         private static readonly tokenSymbolMap = KrakenAdapter.standardTokenSymbolMap
+
+         // Krakens's GTS CA 1P5 fingerprint.
+         readonly _certFingerprint256 =
+           '97:D4:20:03:E1:32:55:29:46:09:7F:20:EF:95:5F:5B:1C:D5:70:AA:43:72:D7:80:03:3A:65:EF:BE:69:75:8D'
+
+         protected generatePairSymbol(): string {
+           const base = KrakenAdapter.tokenSymbolMap.get(this.config.baseCurrency)
+           const quote = KrakenAdapter.tokenSymbolMap.get(this.config.quoteCurrency)
+           return `${base}${quote}`
+         }
+
+         async fetchTicker(): Promise<Ticker> {
+           const json = await this.fetchFromApi(
+             ExchangeDataType.TICKER,
+             `0/public/Ticker?pair=${this.pairSymbol}`
+           )
+           return this.parseTicker(json)
+         }
+
+         async fetchTrades(): Promise<Trade[]> {
+           // Trade data is not needed by oracle but is required by the parent class.
+           // This function along with all other functions that are not needed by the oracle will
+           // be removed in a future PR. 
+           // -- @bayological ;) --
+           return []
+         }
+
+         /**
+          * @param json a json object representing the ticker from Kraken's API
+          * Expected format can be seen in the public docs": https://docs.kraken.com/rest/#tag/Market-Data/operation/getTickerInformation
+          *
+          */
+         parseTicker(json: any): Ticker {  
+          const data = json.result[Object.keys(json.result)[0]] 
```

Comment:
So there should only be one ticker object under `result` right? Do you think adding an assert here would be good just in case there's something fishy going on with the api? 

something like `assert (Object.keys(json.result).length == 1)`